src/java.base/share/classes/com/sun/crypto/provider/DHKeyAgreement.java

```diff
@@ -313,6 +313,14 @@ protected int engineGenerateSecret(byte[] sharedSecret, int offset)
         // above, so user can recover w/o losing internal state
         generateSecret = false;
 
+        // No further process if z <= 1 or z == (p - 1).
```

Comment:
You could mention the spec somewhere in this file.  Always nice to have the spec mentioned that we have impl'd.  Up to you.